packages/wasm/src/transform/mod.rs

```diff
@@ -0,0 +1,336 @@
+
+            self.imports.insert("Box".to_string(), "__Box".to_string());
+
+            program.body.insert(0, import_statement);
+        }
+
+        program
+    }
+}
+
+pub struct ReplaceKWithBox<'a> {
+    imports: HashMap<String, String>,
+    ast: AstBuilder<'a>,
+}
+
+impl<'a> ReplaceKWithBox<'a> {
+    pub fn new(imports: HashMap<String, String>, allocator: &'a Allocator) -> Self {
+        Self {
+            imports,
+            ast: AstBuilder::new(allocator),
+        }
+    }
+
+    pub fn process_element(
+        &mut self,
+        elem: &mut JSXElement<'a>,
+        k_alias: &str,
+        box_local_name: &str,
+    ) {
+        if let JSXElementName::MemberExpression(member_expr) = &elem.opening_element.name {
+            if let JSXMemberExpressionObject::Identifier(ident) = &member_expr.object {
+                let html_tag = member_expr.property.name.as_str();
+                let html_tag: &'a str = unsafe { std::mem::transmute(html_tag) }; // FIXME: Remove unsafe transmute
+                if ident.name == k_alias {
+                    self.transform_element(elem, box_local_name, &html_tag);
+                }
+            }
+        }
+    }
+
+    pub fn transform_element(
+        &mut self,
+        elem: &mut JSXElement<'a>,
+        box_local_name: &str,
+        html_tag: &'a str,
+    ) {
+        let as_attribute = helpers::create_as_attribute(&self.ast, html_tag);
+        let kuma_attribute = helpers::create_kuma_default_attr(&self.ast);
+        elem.opening_element.attributes.insert(0, kuma_attribute);
+        elem.opening_element.attributes.insert(0, as_attribute);
+
+        let box_name = if box_local_name == "Box" {
+            "Box"
+        } else {
+            "__Box"
+        };
+
+        elem.opening_element.name = JSXElementName::Identifier(self.ast.alloc(JSXIdentifier {
+            span: SPAN,
+            name: Atom::from(box_name),
+        }));
+
+        if let Some(closing_element) = &mut elem.closing_element {
+            closing_element.name = JSXElementName::Identifier(self.ast.alloc(JSXIdentifier {
+                span: SPAN,
+                name: Atom::from(box_name),
+            }));
+        }
+    }
+}
+
+impl<'a> VisitMut<'a> for ReplaceKWithBox<'a> {
+    /**
+     * Processes the JSXElement nodes in the AST and replaces the `k` syntax from `@kuma-ui/core`
+     * with corresponding `Box` component. This allows usage of the `k` syntax as a shorthand for creating
+     * styled components, e.g. `<k.div>` is transformed to `<Box as="div">`.
+     */
+    fn visit_jsx_element(&mut self, elem: &mut JSXElement<'a>) {
+        if let Some(k_alias) = self.imports.get("k").cloned() {
+            if let Some(box_local_name) = self.imports.get("Box") {
+                let k_alias = k_alias.clone();
+                self.process_element(elem, &k_alias, &box_local_name.clone());
+            }
+        }
```

Comment:
https://github.com/oxc-project/oxc/blob/5081652bc10cdcfd10cf57bf0385ebc7eacd5769/crates/oxc_ast/src/visit/visit_mut.rs#L404

You need to call `walk_jsx_element_mut` otherwise it's not going to visit its children.

Reply:
@Boshen 
You are right, thank you so much for taking a look at the PR 🤝